pkgs/development/ocaml-modules/mirage-crypto/default.nix

```diff
@@ -17,12 +20,20 @@ buildDunePackage rec {
   checkInputs = [ ounit ];
 
   nativeBuildInputs = [ dune-configurator pkg-config ];
-  propagatedBuildInputs = [ cstruct eqaf ];
+  propagatedBuildInputs = [
+    cstruct eqaf
+  ] ++ lib.optionals withFreestanding [
+    ocaml-freestanding
+  ];
 
   meta = with lib; {
     homepage = "https://github.com/mirage/mirage-crypto";
     description = "Simple symmetric cryptography for the modern age";
-    license = licenses.isc;
+    license = [
+      licenses.isc  # default license
+      licenses.bsd2 # mirage-crypto-rng-mirage
+      licenses.mit  # mirage-crypto-ec
+    ];
```

Comment:
Why do you list all three licenses here rather than setting the `meta.license` attribute in the relevant packages?

Reply:
Mainly because I am not confident you can just split the licenses up between subpackages if they all come from a single tarball, but upstream seems to believe so as they only list a single license in each opam file. The licenses are at least all compatible with each other…

Reply:
Why are we not using with licenses here? It would remove the repition of licenses.